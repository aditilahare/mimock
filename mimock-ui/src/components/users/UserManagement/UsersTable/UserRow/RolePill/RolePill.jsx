import React from 'react';
import { PropTypes } from 'prop-types';
import { Pill } from './RolePill.style';

export default function RolePill({ role }) {
	return (
		<Pill data-testid={`role-pill-${role}`} role={role}>
			{role}
		</Pill>
	);
}

RolePill.propTypes = {
	role: PropTypes.string.isRequired,
};
