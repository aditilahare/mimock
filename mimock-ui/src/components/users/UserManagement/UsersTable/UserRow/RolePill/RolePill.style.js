import tw from 'tailwind-styled-components';

const roleBackground = {
	ADMIN: 'bg-teal-400',
	MANAGER: 'bg-indigo-400',
	VIEWER: 'bg-slate-500',
};

export const Pill = tw.div`
    font-sans
    font-semibold
    text-white
    text-sm
    text-center
    antialiased
    mx-auto
    w-24
    p-1
    px-2
    rounded-sm
    shadow-sm
    select-none
    
    ${(props) => roleBackground[props.role]}
    ${(props) => (props.$margin ? props.$margin : 'mx-auto')}
`;
