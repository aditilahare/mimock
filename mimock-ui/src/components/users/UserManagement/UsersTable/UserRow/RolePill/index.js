import RolePill from './RolePill';

export default RolePill;
