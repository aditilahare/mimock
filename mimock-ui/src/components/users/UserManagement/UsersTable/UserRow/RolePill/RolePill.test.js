import React from 'react';
import { render, act } from '@testing-library/react';
import RolePill from './RolePill';

describe('RolePill', () => {
	it.each(['ADMIN', 'MANAGER', 'VIEWER'])(
		'should render %s role pill',
		async (role) => {
			let tree;
			await act(async () => {
				tree = await render(<RolePill role={role} />);
			});

			const { container, getByTestId } = tree;

			expect(getByTestId(`role-pill-${role}`)).toBeInTheDocument();

			expect(container).toMatchSnapshot();
		}
	);
});
