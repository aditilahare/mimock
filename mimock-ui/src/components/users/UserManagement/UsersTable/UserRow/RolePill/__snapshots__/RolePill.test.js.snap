// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RolePill should render ADMIN role pill 1`] = `
<div>
  <div
    class="font-sans font-semibold text-white text-sm text-center antialiased mx-auto w-24 p-1 px-2 rounded-sm shadow-sm select-none bg-teal-400"
    data-testid="role-pill-ADMIN"
    role="ADMIN"
  >
    ADMIN
  </div>
</div>
`;

exports[`RolePill should render MANAGER role pill 1`] = `
<div>
  <div
    class="font-sans font-semibold text-white text-sm text-center antialiased mx-auto w-24 p-1 px-2 rounded-sm shadow-sm select-none bg-indigo-400"
    data-testid="role-pill-MANAGER"
    role="MANAGER"
  >
    MANAGER
  </div>
</div>
`;

exports[`RolePill should render VIEWER role pill 1`] = `
<div>
  <div
    class="font-sans font-semibold text-white text-sm text-center antialiased mx-auto w-24 p-1 px-2 rounded-sm shadow-sm select-none bg-slate-500"
    data-testid="role-pill-VIEWER"
    role="VIEWER"
  >
    VIEWER
  </div>
</div>
`;
